DSharpPlus.Commands/CommandsExtension.cs

```diff
@@ -159,6 +172,86 @@ public async ValueTask AddProcessorsAsync(IEnumerable<ICommandProcessor> process
 
     public TProcessor GetProcessor<TProcessor>() where TProcessor : ICommandProcessor => (TProcessor)this._processors[typeof(TProcessor)];
 
+    /// <summary>
+    /// Adds all public checks from the provided assembly to the extension.
+    /// </summary>
+    public void AddChecks(Assembly assembly)
+    {
+        foreach
+        (
+            Type t in assembly.ExportedTypes.Where
+            (
+                candidate => candidate.GetInterfaces().Any
+                (
+                    x => x.FullName == "DSharpPlus.Commands.ContextChecks.IContextCheck`1"
+                )
+            )
+        )
+        {
+            this.AddCheck(t);
+        }
```

Comment:
Horrid formatting. Instead of using Linq for this, iterate only once. Additionally use `Assembly.GetTypes()` for consistency in behavior with other methods that iterate assembly types.
```suggestion
        foreach (Type type in assembly.GetTypes())
        {
            if (type.GetInterface("DSharpPlus.Commands.ContextChecks.IContextCheck`1", false) is not null)
            {
                this.AddCommand(type);
            }
        }
```

Reply:
No, we want `GetTypes` as that's what the rest of the extension currently does. If you're looking to change this behavior, bring it up in #lib-dev